File: currency-frontend/src/app/currency.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CurrencyRequest, CurrencyResponse, ResponseInfo } from './currency-info/currency-info.model';

@Injectable({
  providedIn: 'root'
})
export class CurrencyService {
  private baseUrl = 'http://localhost:8080/currencies';

  constructor(private http: HttpClient) { }

  getCurrentCurrencyValue(currencyRequest: CurrencyRequest): Observable<CurrencyResponse> {
    const url = `${this.baseUrl}/get-current-currency-value-command`;
    return this.http.post<CurrencyResponse>(url, currencyRequest);
  }

  getAllRequests(): Observable<ResponseInfo[]> {
    const url = `${this.baseUrl}/requests`;
    return this.http.get<ResponseInfo[]>(url);
  }
}

File: src/main/java/com/radoslawzerek/recruitment_task/service/CurrencyService.java
package com.radoslawzerek.recruitment_task.service;

import com.radoslawzerek.recruitment_task.entity.RequestInfo;
import com.radoslawzerek.recruitment_task.model.CurrencyRequest;
import com.radoslawzerek.recruitment_task.model.CurrencyResponse;

import java.util.List;

public interface CurrencyService {

    CurrencyResponse getCurrentCurrencyValue(CurrencyRequest currencyRequest);

    List<RequestInfo> getAllRequests();
}
